pkg/handler/v2/operatingsystemprofile/operatingsystemprofile.go

```diff
@@ -0,0 +1,171 @@
+// KKP dashboard.
+// Namespace is purposefully left empty since we cannot determine namespace of these resources before cluster creation.
+var defaultOperatingSystemProfiles = []apiv2.OperatingSystemProfile{
+	{
+		Name: "osp-amzn2",
+	},
+	{
+		Name: "osp-centos",
+	},
+	{
+		Name: "osp-flatcar",
+	},
+	{
+		Name: "osp-rhel",
+	},
+	{
+		Name: "osp-rockylinux",
+	},
+	{
+		Name: "osp-sles",
+	},
+	{
+		Name: "osp-ubuntu",
+	},
+}
+
+// listOperatingSystemProfilesReq defines HTTP request for listOperatingSystemProfilesForCluster
+// swagger:parameters listOperatingSystemProfilesForCluster
+type listOperatingSystemProfilesReq struct {
+	common.ProjectReq
+	// in: path
+	ClusterID string `json:"cluster_id"`
+}
+
+// GetSeedCluster returns the SeedCluster object.
+func (req listOperatingSystemProfilesReq) GetSeedCluster() apiv1.SeedCluster {
+	return apiv1.SeedCluster{
+		ClusterID: req.ClusterID,
+	}
+}
+
+func DecodeListOperatingSystemProfiles(c context.Context, r *http.Request) (interface{}, error) {
+	var req listOperatingSystemProfilesReq
+
+	clusterID, err := common.DecodeClusterID(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.ClusterID = clusterID
+
+	projectReq, err := common.DecodeProjectRequest(c, r)
+	if err != nil {
+		return nil, err
+	}
+	req.ProjectReq = projectReq.(common.ProjectReq)
+
+	return req, nil
+}
+
+func ListOperatingSystemProfilesEndpointForCluster(userInfoGetter provider.UserInfoGetter, operatingSystemProfile provider.OperatingSystemProfileProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		req := request.(listOperatingSystemProfilesReq)
+
+		userClusterNamespace, err := clusterNamespaceFromContext(ctx, userInfoGetter, req.ProjectID, req.ClusterID)
+		if err != nil {
+			return nil, err
+		}
+
+		ospList, err := operatingSystemProfile.ListUnsecured(ctx, userClusterNamespace)
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		var resp []*apiv2.OperatingSystemProfile
+		for _, osp := range ospList.Items {
+			ospModel := &apiv2.OperatingSystemProfile{
+				Name: osp.Name,
+			}
+			resp = append(resp, ospModel)
+		}
+
+		return resp, nil
+	}
+}
+
+func ListOperatingSystemProfilesEndpoint(seedsGetter provider.SeedsGetter, operatingSystemProfile provider.OperatingSystemProfileProvider) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		seeds, err := seedsGetter()
+		if err != nil {
+			return nil, common.KubernetesErrorToHTTPError(err)
+		}
+
+		// Fetch OSPs for all the seeds.
+		var osps []osmv1alpha1.OperatingSystemProfile
+		for _, seed := range seeds {
+			ospList, err := operatingSystemProfile.ListUnsecured(ctx, seed.Namespace)
+			if err != nil {
+				return nil, common.KubernetesErrorToHTTPError(err)
+			}
+
+			osps = append(osps, ospList.Items...)
+		}
```

Comment:
The seeds will all have the same `kubermatic` namespace. And with this you will only get the OSP from the master cluster, which I guess is not your goal?

I understand that the custom OSPs are on the `kubermatic` namspace of the Seed clusters. So you would need to use the dynamic client that gets set based on the Seed in question. For that you would also need the {seed} in the endpoint.

Or if you want to take the custom OSPs from the master cluster, you dont need to iterate over the seeds, you can just get them from the `kubermatic` namespace. But if the OSPs reside in the master cluster, you will also need a synchronizer between master and seed clusters, which syncs the OSPs from master to all seeds (there is a bunch of similar resources already doing the same).

Reply:
I'll properly look into this tomorrow. I think I went forward with the PR in a bit of haste. Which was wrong from my side.

Reply:
Dont worry, we can discuss it tomorrow :)